contracts/extensions/VotingReputation.sol

```diff
@@ -442,7 +447,24 @@ contract VotingReputation is DSMath, PatriciaTreeProofs {
     poll.skillId = newDomainSkillId;
     poll.skillRep = getReputationFromProof(_pollId, address(0x0), _key, _value, _branchMask, _siblings);
 
+    if (poll.votes[NAY] < poll.votes[YAY]) {
+      poll.stakes[NAY] = sub(poll.stakes[NAY], poll.paidVoterComp);
+    } else {
+      poll.stakes[YAY] = sub(poll.stakes[YAY], poll.paidVoterComp);
+    }
```

Comment:
This is a nice solution, and I think has a lot of subtle benefits (if you're escalating, more needs to be staked to overturn the decision already made, but ultimately if you're proved right after multiple back-and-forths through multiple escalations, you get everything back, I think?).

Reply:
Mm I agree that it makes the  cost of overturning high, which is nice, but I don't think you're guaranteed to get everything back if you eventually win, since we don't really know how much stake was paid out to voters along the way. Imagine ten escalations through ten domains each with the same rep -- we might have paid out 100% (10 x 10%) of the stake to voters before reaching the final escalation, meaning that even if I win by a landslide I'll only get 90% of the stake back. I still think this is a good idea since that's an edge case, but it's worth noting.

Reply:
>we might have paid out 100% (10 x 10%)

*Narrows eyes at this maths*

The logic I was leaning on when I made my uncertain statement was that the extra top-up required should be paying for the voter compensation in the previous round, but on closer inspection I don't think the bookkeeping is quite right for that to play out. I think.... `paidVoterComp` needs to be reset to zero once a vote begins? 

* Post-escalation, pre-both-sides sufficiently staking, you don't know that the top up to cover the previous round has occurred, so have to continue to track it in case a vote doesn't happen, and it needs to come out of the previous-round's losers-stake.
* Once a poll starts, you know that top-up has occurred, so you've covered the voter rewards in the previous round and can stop tracking them.

Otherwise you surely start double counting voter rewards, and could even easily start throwing here which would block a poll from being legitimately escalated.

Reply:
Yeah, my original claim patently can't be true. If YAY keeps on losing, but they keep escalating, until they win at the very last vote, you can pay out an arbitrarily large amount in voter rewards, unrelated to the size of the stake on the NAY side that you ended up with which I was claiming would be able to cover it. Which means we have to pay out the voting rewards for a particular vote from whichever side lost the vote, regardless of what happens in the future.

Haven't worked out the consequences of that yet :sweat_smile:

Reply:
I think it means we can't solve it like this. The actual number of tokens staked by users, and this staked total we're tracking get out of sync, which messes up the calculation of `stakeFraction` in `claimReward`.

Reply:
Mm I see what you mean -- if we subtract from `stakes`, we'll get to a place where the sum of everyone's `stakeFraction` is greater than 100%. I think the answer here is a second counter unfortunately, something along the lines of `stakes` and `totalStakes`, the latter of which is used only to calculate `stakeFraction`.

Reply:
After a call, we ended up inverting the problem and storing the `paidVoterStakes` after each round. We had to re-jigger the reward claiming logic to account for this, but it all seems to have come together.